src/Symfony/Contracts/Translation/TranslatorInterface.php

```diff
@@ -52,14 +54,14 @@ interface TranslatorInterface
      *
      * @see https://en.wikipedia.org/wiki/ISO_31-11
      *
-     * @param string      $id         The message id (may also be an object that can be cast to string)
-     * @param array       $parameters An array of parameters for the message
-     * @param string|null $domain     The domain for the message or null to use the default
-     * @param string|null $locale     The locale or null to use the default
+     * @param string|Translatable $id         The message id (may also be an object that can be cast to string)
+     * @param array               $parameters An array of parameters for the message
+     * @param string|null         $domain     The domain for the message or null to use the default
+     * @param string|null         $locale     The locale or null to use the default
      *
      * @return string The translated string
      *
      * @throws \InvalidArgumentException If the locale contains invalid characters
      */
-    public function trans(string $id, array $parameters = [], string $domain = null, string $locale = null);
+    public function trans($id, array $parameters = [], string $domain = null, string $locale = null);
```

Comment:
Isn't that a bc break?

Reply:
:sob: we need to rethink this yes. As we can assume each Translatable can be forwarded to trans(), what about

```
static Translatable::trans(TranslatorInterface $translator, self $translatable) {
   return $translator->trans($translatable->getMessageId(), ...);
}
```

Then at least the twig `trans` function can support string|Translatable

Reply:
I like that idea. This would also allow us to introduce this functionality without really disrupting anything. If it turns out we eventually do want to integrate further, which might not happen, we can add a deprecation warning for the next major release.

I'll rework the PR and request another review.

Reply:
Updated PR